diff --git a/ualpn/ualpn.py b/ualpn/ualpn.py
index d4eccc1..40eb2f6 100644
--- a/ualpn/ualpn.py
+++ b/ualpn/ualpn.py
@@ -4,8 +4,6 @@ import hashlib
 import socket
 import josepy as jose
 
-import zope.interface
-
 from acme import challenges
 #from certbot import achallenges  # pylint: disable=unused-import
 from certbot import errors
@@ -15,9 +13,7 @@ from certbot.plugins import common
 
 logger = logging.getLogger(__name__)
 
-@zope.interface.implementer(interfaces.IAuthenticator)
-@zope.interface.provider(interfaces.IPluginFactory)
-class Authenticator(common.Plugin):
+class Authenticator(common.Plugin, interfaces.Authenticator):
     """TLS-ALPN-01 ualpn authenticator
 
     This authenticator sets up ualpn to fulfill a tls-alpn-01 challenge.
