From afd4be3117456ce3255900e3ac322caeb8fc73f2 Mon Sep 17 00:00:00 2001
From: Steinar Seljevold <steinar+github@dbq.no>
Date: Sun, 8 Jan 2023 16:32:28 +0100
Subject: [PATCH] Update requirements

---
 pyproject.toml         |  6 ++---
 1 files changed, 29 insertions(+), 35 deletions(-)

diff --git ./pyproject.toml ./pyproject.toml
index 142a689..61d9c7f 100644
--- a/pyproject.toml
+++ ./pyproject.toml
@@ -12,11 +12,11 @@ dependencies = [
        'appdirs>=1.4.3',
        'pyparsing>=2.2.0',
        'packaging>=16.8',
-       'pyOpenSSL>=17.5.0,<20.0.0',
+       'pyOpenSSL>=17.5.0',
        'py3dns>=3.1.0',
-       'cryptography>=2.1.4,<36.0.0',
+       'cryptography>=2.1.4',
        'asn1crypto>=0.24.0',
-       'acme>=0.25.1,<1.19.0',
+       'acme>=0.25.1',
        'PyYAML>=3.1',
 ]
 dynamic = ['version']
--
2.38.2
